src/Sylius/Component/Storage/CookieStorage.php

```diff
@@ -53,6 +64,12 @@ public function getData($key, $default = null)
     public function setData($key, $value)
     {
         $this->request->cookies->set($key, $value);
+
+        $this->eventDispatcher->addListener('kernel.response', function (FilterResponseEvent $event) use ($key, $value) {
```

Comment:
This is coupling the component to Symfony. Is there any way to avoid that? I'd be fine with coupling it to PSR-7.